pkg/common/patterns.go

```diff
@@ -8,6 +8,7 @@ import (
 )
 
 const EmailPattern = `\b(?:[a-z0-9!#$%&'*+/=?^_\x60{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_\x60{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])\b`
+const JWTPattern = `\b[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\b`
```

Comment:
@sahil9001 this is too loose. Take a look at this for a reference https://github.com/gitleaks/gitleaks/blob/43fae355e6fe4d99d2a7b240a224b85e2903aeb4/config/gitleaks.toml#L2311 or even https://github.com/trufflesecurity/trufflehog/blob/6a367ab763b77c47c905bb950a801dbc6bfbe97c/pkg/detectors/formio/formio.go#L23. The first section might be consistent across all bombomb tokens.

Reply:
No, it changes with each refresh, apparently there is a `jti` field which changes each time.
![Screenshot 2024-10-28 at 9 04 41 PM](https://github.com/user-attachments/assets/80b976a0-415f-4247-bfb7-a6869c72e773)

Reply:
@sahil9001 JWT are base64 encoded so that means there will be a common prefix for _all_ JWT, `base64('{"')=ey`. Take a look at https://www.ibm.com/docs/en/cics-ts/6.x?topic=cics-json-web-token-jwt.

Reply:
Hey, thanks, I read more about it, I found out that —

1. The `typ` part is common for all JWT tokens.
2. `alg` and `jti` changes due to which we can only constraint the JWT common prefix upto `ey`.


Please take a look, I have made the change.